rasa/utils/tf_models.py

```diff
@@ -139,7 +143,34 @@ def predict(
     def train_on_batch(
         self, batch_in: Union[Tuple[np.ndarray], Tuple[tf.Tensor]], **kwargs
     ) -> None:
-        raise NotImplementedError
+        with tf.GradientTape() as tape:
+            losses, scores = self._train_losses_scores(batch_in)
+            regularization_loss = tf.math.add_n(self.losses)
+            pred_loss = tf.math.add_n(list(losses.values()))
+            total_loss = pred_loss + regularization_loss
+
+        gradients = tape.gradient(total_loss, self.trainable_variables)
+        self._optimizer.apply_gradients(zip(gradients, self.trainable_variables))
+
+        self.train_metrics["t_loss"].update_state(total_loss)
+        for k, v in losses.items():
+            self.train_metrics[k].update_state(v)
+        for k, v in scores.items():
+            self.train_metrics[k].update_state(v)
+
+    def evaluate_on_batch(
+        self, batch_in: Union[Tuple[np.ndarray], Tuple[tf.Tensor]]
```

Comment:
```suggestion
        self, batch_in: Union[Tuple[np.ndarray], Tuple[tf.Tensor]], **kwargs
```
In order to pass lint check.